compiler_opt/rl/corpus.py

```diff
@@ -0,0 +1,180 @@
+          delete_flags: Tuple[str, ...]) -> List[ModuleSpec]:
+    """Fetch a list of ModuleSpecs for the corpus at data_path
+
+    Args:
+      data_path: base directory of corpus
+      additional_flags: tuple of clang flags to add.
+      delete_flags: tuple of clang flags to remove.
+    """
+    raise NotImplementedError
+
+
+def _has_thinlto_index(module_paths: Iterable[str]) -> bool:
+  return tf.io.gfile.exists(next(iter(module_paths)) + '.thinlto.bc')
+
+
+def _has_cmd(module_paths: Iterable[str]) -> bool:
+  return tf.io.gfile.exists(next(iter(module_paths)) + '.cmd')
+
+
+def _load_module_paths(data_path, module_paths_path: str) -> List[str]:
+  with open(module_paths_path, 'r', encoding='utf-8') as f:
+    ret = [os.path.join(data_path, name.rstrip('\n')) for name in f]
+    if len(ret) == 0:
+      logging.fatal('%s is empty.', module_paths_path)
+      raise ValueError
+    return ret
+
+
+def _load_metadata(metadata_path: str) -> Dict[any, any]:
+  try:
+    with open(metadata_path, 'r', encoding='utf-8') as f:
+      return json.load(f)
+  except FileNotFoundError:
+    logging.info('%s couldn\'t be found.', metadata_path)
+    return {}
+
+
+def _load_and_parse_command(cmd_file: Optional[str],
+                            ir_file: str,
+                            thinlto_file: Optional[str] = None,
+                            additional_flags: Tuple[str, ...] = (),
+                            delete_flags: Tuple[str, ...] = (),
+                            cmd_override: Tuple[str] = None) -> List[str]:
+  """Loads and cleans up base command line.
+
+  Remove certain unnecessary flags, and add the .bc file to compile and, if
+  given, the thinlto index.
+
+  Args:
+    cmd_file: Path to a .cmd file (from corpus).
+    ir_file: The path to the ir file to compile.
+    thinlto_file: The path to the thinlto index, or None.
+    additional_flags: Tuple of flags to add.
+    delete_flags: Tuple of flags to remove.
+
+  Returns:
+    The argument list to pass to the compiler process.
+  """
+  cmdline = []
+  option = None
+  if cmd_override is not None:
+    cmdline = list(cmd_override)
+  elif cmd_file is not None:
+    with open(cmd_file, encoding='utf-8') as f:
+      option_iterator = iter(f.read().split('\0'))
+      option = next(option_iterator, None)
+  else:
+    logging.fatal('.cmd file not available and no command override specified.')
+    raise ValueError
+
+  while option:
+    if any(option.startswith(flag) for flag in delete_flags):
+      if '=' not in option:
+        next(option_iterator, None)
+    else:
+      cmdline.append(option)
+    option = next(option_iterator, None)
+
+  cmdline.extend(['-x', 'ir', ir_file])
+
+  if thinlto_file:
+    cmdline.append('-fthinlto-index=' + thinlto_file)
+    cmdline += ['-mllvm', '-thinlto-assume-merged']
+
+  if cmd_override is None:
+    cmdline.extend(additional_flags)
+    # Ensure that -cc1 is always present
+    if cmdline[0] != '-cc1':
```

Comment:
is it supposed to happen?

Reply:
Only considering non-linker extracted corpora, i.e. when there's `.cmd` files, the command line inside the `.cmd` file should be run with `-cc1`. Will switch it to `if cmd_file is not None` though, safer that way